students/DimaSakovich/02-structural-linguistics/3-collocations.ipynb

```diff
@@ -0,0 +1,2864 @@
+       "\n",
+       "Підказка: можна шукати послідовність прикметника та іменника і визначати словосполучення через pymorphy2. Також можна попрацювати з деревами залежностей через StanfordNLP, але можливо доведеться запускати код десь на Google Colab, бо там дуже великі моделі для української мови.\n",
+       "\n",
+       "### Джерела\n",
+       "\n",
+       "Ви можете використати будь-яку мову програмування та будь-яку NLP-бібліотеку.\n",
+       "\n",
+       "Корпус блогів взятий з [Political Blog Corpora](http://www.cs.cmu.edu/~ark/blog-data/).\n"
+      ],
+      "text/plain": [
+       "<IPython.core.display.Markdown object>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "show_markdown(os.path.join(TASK_PATH, \"3-collocations.md\"))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Part 1"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import re\n",
+    "import gc\n",
+    "import json\n",
+    "import spacy\n",
+    "import pandas as pd\n",
+    "\n",
+    "from collections import Counter, OrderedDict\n",
+    "from functools import reduce\n",
+    "from multiprocessing import cpu_count\n",
+    "from tqdm import tqdm\n",
+    "from tqdm.contrib.concurrent import process_map"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "nlp_en = spacy.load('en_core_web_md')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "synonyms = [\"say\", \n",
+    "            \"add\", \"announce\", \"answer\", \"assert\", \"claim\", \"convey\", \"declare\", \"deliver\",\n",
+    "            \"disclose\", \"do\", \"estimate\", \"express\", \"maintain\", \"mention\", \"read\", \"repeat\", \"reply\",\n",
+    "            \"report\", \"respond\", \"reveal\", \"speak\", \"state\", \"suggest\", \"tell\", \"voice\", \"affirm\", \n",
+    "            \"allege\", \"communicate\", \"conjecture\", \"divulge\", \"flap\", \"gab\", \"guess\", \"imagine\", \n",
+    "            \"imply\", \"jaw\", \"judge\", \"lip\", \"opine\", \"orate\", \"perform\", \"pronounce\", \"rap\", \"recite\", \n",
+    "            \"rehearse\", \"relate\", \"remark\", \"render\", \"rumor\", \"spiel\", \"utter\", \"verbalize\", \"yak\"]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def count_adv(sentence, verbs=synonyms):\n",
+    "    tokens = nlp_en(sentence)\n",
+    "    cnt = Counter()\n",
+    "    \n",
+    "    for token in tokens:\n",
+    "        if token.pos_ == 'ADV' and token.text[-2:] == 'ly' and token.dep_ == 'advmod':\n",
+    "            for ancestor in token.ancestors:\n",
```

Comment:
В той час як інші студенти зібрали надто мало синонімів, у тебе їх трохи забагато 😄Пошук по всіх предках дістає купу випадків, де дієслово і прислівник не є пов'язані напряму. Наприклад, твій код дістане "tell seriously" з цього речення:

_She told me that her daughter was in danger because someone seriously threatened her._

Це вже не колокація в такому випадку.

Reply:
сьогодні виправлю)

Reply:
поставив вибір тільки по найближчому предку